extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/tunnel/RestrictedGuacamoleTunnelService.java

```diff
@@ -179,9 +187,27 @@ protected ModeledConnection acquire(RemoteAuthenticatedUser user,
 
             @Override
             public int compare(ModeledConnection a, ModeledConnection b) {
-
-                return getActiveConnections(a).size()
-                     - getActiveConnections(b).size();
+                
+                int weightA, weightB;
+                // Check if weight of a is non-null and retrieve it.
+                if (a.getConnectionWeight() != null && a.getConnectionWeight().intValue() > 0)
+                    weightA = a.getConnectionWeight().intValue();
+                // In all other cases assign 1 for sorting.
+                else
+                    weightA = 1;
+
+                // Check if weight of b is null, assign 1 if it is.
+                if (b.getConnectionWeight() != null && b.getConnectionWeight().intValue() > 0)
+                    weightB = b.getConnectionWeight().intValue();
+                // In all other cases assign 1 for sorting.
+                else
+                    weightB = 1;
+
+                // Get current active connections, add 1 to both to avoid calculations with 0.
+                int connsA = getActiveConnections(a).size() + 1;
+                int connsB = getActiveConnections(b).size() + 1;
+
+                return (connsA * 10000 / weightA) - (connsB * 10000 / weightB);
```

Comment:
There's no significance to the 10000 itself, that number is arbitrary.  The overall idea, though, is to get the values high enough that the final calculation returns an integer.  Because the number of connections is going to be a relatively low number (between 1 and 100, maybe 1000), and the weights can be relatively high (100, 1000, 10000, etc.), the idea is to get the numerator sufficiently high so that an integer calculation will actually happen.